test/screenshot.spec.js

```diff
@@ -39,19 +39,6 @@ module.exports.addTests = function({testRunner, expect, product}) {
       });
       expect(screenshot).toBeGolden('screenshot-clip-rect.png');
     });
-    it('should work for offscreen clip', async({page, server}) => {
```

Comment:
this looks like a pretty huge change. Are we sure that its ok? At the very least it appears to be a breaking change. It also doesn't appear to match the Firefox behavior now.

Reply:
I learned that there's some history here: https://bugs.chromium.org/p/chromium/issues/detail?id=937609#c4 We'll call it out as a breaking change in the v2 release. The merge commit mentions the following:

> Per upstream Chromium changes, `page.screenshot` now clips elements to the viewport. This matches the clipping behavior of elements in inner scrollers (i.e., document and overflow scroll clipping now work the same).

This seems like a useful change.

Reply:
If there is new intended behavior, I'd expect to see a new test.

Reply:
Is it possible to remove the restriction on that crbug? I can't see it.

Reply:
Unfortunately not, but I gave you access before posting my previous comment.

Reply:
> If there is new intended behavior, I'd expect to see a new test.

Sounds reasonable. Let's restore the test, update the expectation, and mark it as failing in Firefox.

Reply:
https://github.com/GoogleChrome/puppeteer/pull/5079